L1Trigger/L1TTrackMatch/plugins/L1TkMuonProducer.cc

```diff
@@ -0,0 +1,712 @@
+  std::unique_ptr<L1TkMuMantra> mantracorr_endc_;
+  int mantra_n_trk_par_;
+
+  const edm::EDGetTokenT<RegionalMuonCandBxCollection> bmtfToken;
+  const edm::EDGetTokenT<RegionalMuonCandBxCollection> omtfToken;
+  const edm::EDGetTokenT<RegionalMuonCandBxCollection> emtfToken;
+  const edm::EDGetTokenT<EMTFTrackCollection>
+      emtfTCToken;  // the track collection, directly from the EMTF and not formatted by GT
+  const edm::EDGetTokenT<std::vector<TTTrack<Ref_Phase2TrackerDigi_> > > trackToken;
+};
+
+L1TkMuonProducer::L1TkMuonProducer(const edm::ParameterSet& iConfig)
+    : bmtfToken(consumes<RegionalMuonCandBxCollection>(iConfig.getParameter<edm::InputTag>("L1BMTFInputTag"))),
+      omtfToken(consumes<RegionalMuonCandBxCollection>(iConfig.getParameter<edm::InputTag>("L1OMTFInputTag"))),
+      emtfToken(consumes<RegionalMuonCandBxCollection>(iConfig.getParameter<edm::InputTag>("L1EMTFInputTag"))),
+      emtfTCToken(consumes<EMTFTrackCollection>(iConfig.getParameter<edm::InputTag>("L1EMTFTrackCollectionInputTag"))),
+      trackToken(consumes<std::vector<TTTrack<Ref_Phase2TrackerDigi_> > >(
+          iConfig.getParameter<edm::InputTag>("L1TrackInputTag"))) {
+  ETAMIN_ = (float)iConfig.getParameter<double>("ETAMIN");
+  ETAMAX_ = (float)iConfig.getParameter<double>("ETAMAX");
+  ZMAX_ = (float)iConfig.getParameter<double>("ZMAX");
+  CHI2MAX_ = (float)iConfig.getParameter<double>("CHI2MAX");
+  PTMINTRA_ = (float)iConfig.getParameter<double>("PTMINTRA");
+  DRmax_ = (float)iConfig.getParameter<double>("DRmax");
+  nStubsmin_ = iConfig.getParameter<int>("nStubsmin");
+
+  // --- mantra corr params
+
+  mantra_n_trk_par_ = iConfig.getParameter<int>("mantra_n_trk_par");
+
+  // --------------------- configuration of the muon algorithm type
+
+  std::string bmtfMatchAlgoVersionString = iConfig.getParameter<std::string>("bmtfMatchAlgoVersion");
+  std::transform(bmtfMatchAlgoVersionString.begin(),
+                 bmtfMatchAlgoVersionString.end(),
+                 bmtfMatchAlgoVersionString.begin(),
+                 ::tolower);  // make lowercase
+
+  std::string omtfMatchAlgoVersionString = iConfig.getParameter<std::string>("omtfMatchAlgoVersion");
+  std::transform(omtfMatchAlgoVersionString.begin(),
+                 omtfMatchAlgoVersionString.end(),
+                 omtfMatchAlgoVersionString.begin(),
+                 ::tolower);  // make lowercase
+
+  std::string emtfMatchAlgoVersionString = iConfig.getParameter<std::string>("emtfMatchAlgoVersion");
+  std::transform(emtfMatchAlgoVersionString.begin(),
+                 emtfMatchAlgoVersionString.end(),
+                 emtfMatchAlgoVersionString.begin(),
+                 ::tolower);  // make lowercase
+
+  if (bmtfMatchAlgoVersionString == "tp")
+    bmtfMatchAlgoVersion_ = kTP;
+  else if (bmtfMatchAlgoVersionString == "mantra")
+    bmtfMatchAlgoVersion_ = kMantra;
+  else
+    throw cms::Exception("TkMuAlgoConfig") << "the ID of the BMTF algo matcher passed is invalid\n";
```

Comment:
in these messages, probably useful to print the actual value of bmtfMatchAlgoVersionString (etc. for others)

Reply:
done